opapi/src/section-types-generator/section-types-generator.ts

```diff
@@ -0,0 +1,87 @@
+import { initDirectory, saveFile } from 'src/file'
+import {
+  parseFunctionDefinition,
+  parseParameterTypes,
+  parseRequestParameterTypes,
+  parseSectionTypes,
+  parseReturnTypes,
+} from './section-types-generator.parsers'
+import { Block, DefaultState, OperationParser, SectionParser, BlockComposer } from './section-types-generator.types'
+import { camel } from 'radash'
+import { getBlankBlock } from './section-types-generator.helpers'
+
+const operationGenerators: OperationParser[] = [
+  parseFunctionDefinition,
+  parseParameterTypes,
+  parseRequestParameterTypes,
+  parseReturnTypes,
+]
+const sectionGenerators: SectionParser[] = [parseSectionTypes]
+
+/**
+ * Generates files containing typescript types for each item in the state object - Sections, Operations, Responses, etc.
+ */
+export async function generateTypesBySection(state: DefaultState, targetDirectory: string) {
+  initDirectory(targetDirectory)
+  state.sections.forEach(async (section) => {
```

Comment:
**chore:** simple for loops are way better in typescript. Here, if an iteration fails, it will result in an unhandled rejection. Please use the following syntax:

```ts
for (const section of state.sections) { ... }
```